### ISSUE TITLE
`allow-unwrap-in-tests` should allow `unwrap_err`

### ISSUE THREAD

**Issue by erikschul** [rust-lang/rust-clippy#14145]
### Summary

given

```
unwrap_used = "deny"
allow-unwrap-in-tests = "allow"
```

results in

```
used `unwrap_err()` on a `Result` value
if this value is an `Ok`, it will panic
consider using `expect_err()` to provide a better panic message
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unwrap_used
```

but expected to be allowed.

### Reproducer

I tried this code:

```rust
<code>
```

I saw this happen:

```
<output>
```

I expected to see this happen:

### Version

```text

```

**Comment by alex-semenyuk** [comment]
Could you please try 

`allow-unwrap-in-tests = true`

As it tests https://github.com/rust-lang/rust-clippy/blob/0d3bf65bd45435c91a776972b7da8500562c2fe1/tests/ui-toml/unwrap_used/clippy.toml

**Comment by erikschul** [comment]
```
{"clippy":{"unwrap_used":"deny","allow-unwrap-in-tests":"true","print_stdout":"deny","allow_print_in_tests":"allow"}} is not valid under any of the schemas listed in the 'anyOf' keyword
```

given

```
cargo clippy --version
clippy 0.1.84 (9fc6b43126 2025-01-07)
```

but running `cargo clippy` also gives these warnings which is weird:

```
warning[E0602]: unknown lint: `clippy::allow_print_in_tests`
  |
  = help: did you mean: `clippy::allow_attributes`
  = note: requested on the command line with `-A clippy::allow_print_in_tests`
  = note: `#[warn(unknown_lints)]` on by default

warning[E0602]: unknown lint: `clippy::allow_unwrap_in_tests`
  |
  = note: requested on the command line with `-A clippy::allow_unwrap_in_tests`

```

I also tried using `clippy.toml` but that seems to have very limited support:

```
error: error reading Clippy's configuration file: unknown field `unwrap_used`, expected one of
           absolute-paths-allowed-crates             enum-variant-size-threshold
           absolute-paths-max-segments               excessive-nesting-threshold
           accept-comment-above-attributes           future-size-threshold
           accept-comment-above-statement            ignore-interior-mutability
           allow-comparison-to-zero                  large-error-threshold
           allow-dbg-in-tests                        literal-representation-threshold
           allow-expect-in-tests                     matches-for-let-else
           allow-mixed-uninlined-format-args         max-fn-params-bools
           allow-one-hash-in-raw-strings             max-include-file-size
           allow-panic-in-tests                      max-struct-bools
           allow-print-in-tests                      max-suggested-slice-pattern-length
           allow-private-module-inception            max-trait-bounds
           allow-renamed-params-for                  min-ident-chars-threshold
           allow-unwrap-in-tests                     missing-docs-in-crate-items
           allow-useless-vec-in-tests                module-item-order-groupings
           allowed-dotfiles                          msrv
           allowed-duplicate-crates                  pass-by-value-size-limit
           allowed-idents-below-min-chars            pub-underscore-fields-behavior
           allowed-prefixes                          semicolon-inside-block-ignore-singleline
           allowed-scripts                           semicolon-outside-block-ignore-multiline
           allowed-wildcard-imports                  single-char-binding-names-threshold
           arithmetic-side-effects-allowed           source-item-ordering
           arithmetic-side-effects-allowed-binary    stack-size-threshold
           arithmetic-side-effects-allowed-unary     standard-macro-braces
           array-size-threshold                      struct-field-name-threshold
           avoid-breaking-exported-api               suppress-restriction-lint-in-const
           await-holding-invalid-types               third-party
           blacklisted-names                         too-large-for-stack
           cargo-ignore-publish                      too-many-arguments-threshold
           check-private-items                       too-many-lines-threshold
           cognitive-complexity-threshold            trait-assoc-item-kinds-order
           cyclomatic-complexity-threshold           trivial-copy-size-limit
           disallowed-macros                         type-complexity-threshold
           disallowed-methods                        unnecessary-box-size
           disallowed-names                          unreadable-literal-lint-fractions
           disallowed-types                          upper-case-acronyms-aggressive
           doc-valid-idents                          vec-box-size-threshold
           enable-raw-pointer-heuristic-for-send     verbose-bit-mask-threshold
           enforce-iter-loop-reborrow                warn-on-all-wildcard-imports
           enforced-import-renames                   warn-unsafe-macro-metavars-in-private-macros
           enum-variant-name-threshold  
```

**Comment by erikschul** [comment]
Hmm, but by using both files, it kind of works:

cargo.toml:
```
[lints.clippy]
unwrap_used = "deny"
```

clippy.toml:
```
allow-unwrap-in-tests = true
```

although... without clippy.toml, `cargo clippy` doesn't throw an error, but VSCode (rust-analyzer) complains. VSCode doesn't complain when I add `clippy.toml`.
Shouldn't `cargo clippy` fail always?
